Add units to timeout argument in EventFlags

kegilbert opened the pull request:
### Description

Add units to the timeout argument in wait_all/any:

https://github.com/ARMmbed/mbed-os/issues/8894


### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [X] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

@sarahmarshy 
@AnotherButler

kjbracey commented:
This phrasing occurs in quite a lot of other RTOS header files, as they were a copy-and-paste from CMSIS-RTOS docs (but might be CMSIS-RTOS v1 rather than v2?). Would be worth updating the others if doing `EventFlags`, as I think consistency is important here - they all have exactly the same time concept.

I can immediately think of `Thread`, `ThisThread`, `Semaphore`, `Mutex`, possibly `ConditionVariable`.

As a sort of compromise, some of those seem to have kept the phrasing, but renamed the `timeout` variable to `millisec`.

kegilbert commented:
@kjbracey-arm It looks like everything else in rtos uses the millisecond parameter name with the old description. I'm not as much of a fan of this since I find it's a bit strange to name the variable after the units and not what it does but as everything uses that I could instead change the EventFlag parameter to millisecond for consistency.

0Grit commented:
#9077 

Edit: Lightly related issue

cmonr commented:
@loverdeg-ep Mind elaborating on your single-link comment?

0Grit commented:
> @loverdeg-ep Mind elaborating on your single-link comment?

Kind of a personal note I suppose. Like to be able to retrace interesting & related conversations for my future self.

I'll cut it out if causing trouble.

kjbracey reviewed (changes requested):
As per comment above, I'm going to request that this be made consistent across the `rtos` classes.

I don't mind whether you change this to match the others, or change all the others to match your new form here - others can express a preference.

cmonr commented:
> I'll cut it out if causing trouble.

Nah, no need. That's fine.

kegilbert committed:
Change EventFlag timeout paramter

Matches rest of RTOS class timeout parameters by using the unit name. Remove ambigious statement in reference to 0 ms being no-timeout as a timeout of 0 causes the function to not block and return immediately (osWaitForever is used as no timeout as it will wait forever)
